cmd/tools/rpcwrappers/main.go

```diff
@@ -111,6 +114,20 @@ var (
 		"metricsClient.matching.PollWorkflowTaskQueue": true,
 		"metricsClient.matching.QueryWorkflow":         true,
 	}
+	// Fields to ignore when looking for the routing fields in a request object.
+	ignoreField = map[string]bool{
```

Comment:
I think we're getting into some sketchy territory here with this level of reflection. If someone forgets to add a new request here, and the wrong routing field gets extracted, it will be really difficult to troubleshoot, and I don't think our tests will catch it. I think this stuff should just be in code instead of string configs. In essence, we can just write the code that extracts the routing field.

Reply:
That's exactly what this PR is fixing

Reply:
Can you sketch out what "code that extracts the routing field" would look like and how it would be called? The goal here was to make it so that most of the time, adding new RPCs required just writing a new handler and not much else (rate limiter quota config maybe). If we have to write code for each request, that kinda defeats the purpose

Reply:
> That's exactly what this PR is fixing

For one instance, but it doesn't future proof us for new RPCs.

> Can you sketch out what "code that extracts the routing field" would look like

I think we can add a call to something like `getXXXShardingKey` to the generated code, and then it will throw a compiler error if that is unimplemented.

> If we have to write code for each request, that kinda defeats the purpose

I think there's still a lot of code that will be auto-generated like the metrics, retries and redirects--just determining the sharding key won't be auto-gen'd.

Reply:
> For one instance, but it doesn't future proof us for new RPCs.

That's what it does, it causes it to error if there's ambiguous routing for new RPCs.

> I think we can add a call to something like `getXXXShardingKey` to the generated code, and then it will throw a compiler error if that is unimplemented.

What is XXX? Do I have to write that for each request object?

There are more exceptions than before, so it may be worth revisiting this, but given that the heuristics work for.. 83% of calls, and with this they do cause errors on ambiguity or lack of key, I think adding back a bunch of hand-written code would be a step backwards.

Reply:
> error if more than one field with the desired name is found.

My bad, I totally missed this. I think with this in place, the possibility of us using the wrong routing key is really low.

Reply:
I realized I can improve this a little more, to catch if there are multiple potential routing fields possible for history